packages/core/ui-settings/core-ui-settings-server-internal/src/clients/ui_settings_user_client.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { KibanaRequest } from '@kbn/core-http-server';
+import type { IUserUiSettingsClient } from '@kbn/core-ui-settings-server/src';
+import type { UiSettingsServiceOptions } from '../types';
+import { BaseUiSettingsClient } from './base_ui_settings_client';
+
+interface UserProvidedValue<T = unknown> {
+  userValue?: T;
+  isOverridden?: boolean;
+}
+
+type UserProvided<T = unknown> = Record<string, UserProvidedValue<T>>;
+
+/**
+ * Common logic for setting / removing keys in a {@link IUserUiSettingsClient} implementation
+ */
+export class UiSettingsUserClient extends BaseUiSettingsClient implements IUserUiSettingsClient {
+  private readonly userProfileSettingsClient: UiSettingsServiceOptions['userProfileSettingsClient'];
+
+  constructor(options: UiSettingsServiceOptions) {
+    super(options);
+    const { userProfileSettingsClient } = options;
+    this.userProfileSettingsClient = userProfileSettingsClient;
+  }
+
+  async getUserProfileSettings(request: KibanaRequest): Promise<Record<string, string>> {
+    return (await this.userProfileSettingsClient?.get(request)) || {};
+  }
+
+  async getUserProvided<T = unknown>(): Promise<UserProvided<T>> {
+    this.log.warn('`getUserProvided` operation is not supported for User Settings.');
+    return {};
+  }
+
+  async setMany(changes: Record<string, any>) {
+    this.log.warn('`setMany` operation is not supported for User Settings.');
+  }
```

Comment:
What's the purpose of trying to replicate the same interface as the other UISettings clients then?

Reply:
Based on the early design concerns, I wanted to so stub a path so that if we did need to build out user settings stored/migrated to SO, it would be ready to be filled out.

Maybe this makes more sense with my suggestion to align all the client interfaces [here](https://github.com/elastic/kibana/pull/151507#issuecomment-1487845604)?